### PR TITLE
fix(core): cap Horde actor-critic array-loop sequence length before scan hang

### PR DESCRIPTION
**Pull request by rama0x1**
## Problem

`run_horde_actor_critic_from_arrays` and `run_nonlinear_horde_actor_critic_from_arrays` hand their pre-collected step arrays straight to `jax.lax.scan` with no bound on the leading (step) axis and no check that the step arrays share a length. A hostile or mistaken caller supplying a huge array forces JAX to trace/compile a scan of that length, hanging the process well before any step executes.

Same hang class already fixed elsewhere in `core/`: `sarsa.py` (#1996), `average_reward.py` (#1999), `partner_policy_fusion.py` (#1991), `learners.py` (#1989), `utils/nexting.py` (#1992) — `horde_actor_critic.py` was not covered by any of those PRs.

## Fix

Adds `_require_horde_ac_sequence_length` (exact-type array + shape gate, rejects length outside `[1, 10_000]` before any scan) and `_require_horde_ac_matching_length` (checks `observations`/`next_observations`/`auxiliary_cumulants`/`discounts` share `rewards`' leading length). The 10,000 ceiling matches the established convention in this module family.

## Verification

- `pytest tests/test_horde_actor_critic.py -q` — 87 passed (includes new failing-test-first cases with a `jax.lax.scan` monkeypatch spy proving scan never runs on rejected input).
- `ruff check` — all checks passed.

## Collision check

`gh pr list --repo elizaOS/asi --state open --json number,title,files` re-run immediately before this push confirmed no open PR touches `alberta_framework/core/horde_actor_critic.py`.

## Attribution

- Client: claude-code
- Provider: anthropic
- Model: claude-sonnet-5
- Lane: rama0x1-asi-claude-worker-2

---

### CODE REVIEW

**Comment by lalalune** [comment]
Superseded by merged PR #2060, which consolidates this resource boundary with shared scan/work budgets, combined-work checks, and the broader regression panel.
